pkg/kubectl/cmd/auth/cani.go

```diff
@@ -240,3 +275,63 @@ func (o *CanIOptions) resourceFor(mapper meta.RESTMapper, resourceArg string) sc
 
 	return gvr
 }
+
+func (o *CanIOptions) printStatus(status authorizationv1.SubjectRulesReviewStatus) error {
+	if status.Incomplete {
+		fmt.Fprintf(o.ErrOut, "warning: the list may be incomplete: %v\n", status.EvaluationError)
+	}
+
+	breakdownRules := []rbacv1.PolicyRule{}
+	for _, rule := range convertToPolicyRule(status) {
+		breakdownRules = append(breakdownRules, validation.BreakdownRule(rule)...)
+	}
+
+	compactRules, err := validation.CompactRules(breakdownRules)
+	if err != nil {
+		return err
+	}
+	sort.Stable(rbacv1helpers.SortableRuleSlice(compactRules))
+
+	w := printers.GetNewTabWriter(o.Out)
+	defer w.Flush()
+
+	return printAll(w, compactRules)
+}
+
+func convertToPolicyRule(status authorizationv1.SubjectRulesReviewStatus) []rbacv1.PolicyRule {
+	ret := []rbacv1.PolicyRule{}
+	for _, resource := range status.ResourceRules {
+		ret = append(ret, rbacv1.PolicyRule{
+			Verbs:         resource.Verbs,
+			APIGroups:     resource.APIGroups,
+			Resources:     resource.Resources,
+			ResourceNames: resource.ResourceNames,
+		})
+	}
+
+	for _, nonResource := range status.NonResourceRules {
+		ret = append(ret, rbacv1.PolicyRule{
+			Verbs:           nonResource.Verbs,
+			NonResourceURLs: nonResource.NonResourceURLs,
+		})
+	}
+
+	return ret
+}
+
+func printAll(out io.Writer, rules []rbacv1.PolicyRule) error {
+	if _, err := fmt.Fprintf(out, "Resources\tNon-Resource URLs\tResource Names\tVerbs\n"); err != nil {
```

Comment:
This is not how you should print, have a look at https://github.com/kubernetes/kubernetes/blob/113872798d34daa32277616b99a368244f1f7b65/pkg/kubectl/cmd/config/get_contexts.go#L142-L155